aws/resource_aws_sagemaker_lifecycle_configuration.go

```diff
@@ -0,0 +1,168 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsSagemakerLifeCycleConfiguration() *schema.Resource {
```

Comment:
Its unfortunately verbose, but we should probably follow the API naming here for the resource functions and resource name to clarify this is specifically for managing Notebook Instance Lifecycle Configurations. This is in case the SageMaker API introduces other "Lifecycle Configurations".

```
aws_sagemaker_notebook_instance_lifecycle_configuration
```

```suggestion
func resourceAwsSagemakerNotebookInstanceLifeCycleConfiguration() *schema.Resource {
```